### ISSUE TITLE
TestNotifierSendsUserIDHeader timed out

### ISSUE THREAD

**Issue by tomwilkie** [weaveworks/cortex#428]
```
panic: test timed out after 1m0s



goroutine 83 [running]:

testing.startAlarm.func1()

	/usr/local/go/src/testing/testing.go:1023 +0xf9

created by time.goFunc

	/usr/local/go/src/time/sleep.go:170 +0x44



goroutine 1 [chan receive]:

testing.(*T).Run(0xc4205a8c30, 0x1d091dd, 0x1d, 0x1d64bc8, 0xc4205ebd20)

	/usr/local/go/src/testing/testing.go:698 +0x2f4

testing.runTests.func1(0xc4205a8c30)

	/usr/local/go/src/testing/testing.go:882 +0x67

testing.tRunner(0xc4205a8c30, 0xc4205ebde0)

	/usr/local/go/src/testing/testing.go:657 +0x96

testing.runTests(0xc4201a5860, 0x2997360, 0xb, 0xb, 0x1825d92)

	/usr/local/go/src/testing/testing.go:888 +0x2c1

testing.(*M).Run(0xc4205ebf20, 0xc4205ebf20)

	/usr/local/go/src/testing/testing.go:822 +0xfc

main.main()

	github.com/weaveworks/cortex/pkg/ruler/_test/_testmain.go:62 +0xf7



goroutine 17 [syscall, 1 minutes, locked to thread]:

runtime.goexit()

	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1



goroutine 11 [chan receive]:

github.com/weaveworks/cortex/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x29a2e00)

	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:882 +0x7a

created by github.com/weaveworks/cortex/vendor/github.com/golang/glog.init.1

	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:410 +0x21d



goroutine 58 [select]:

github.com/weaveworks/cortex/pkg/util.(*hashBucketHistogram).loop(0xc42013b7c0)

	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:87 +0x1e5

created by github.com/weaveworks/cortex/pkg/util.NewHashBucketHistogram

	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:64 +0x18a



goroutine 77 [semacquire]:

sync.runtime_Semacquire(0xc42036154c)

	/usr/local/go/src/runtime/sema.go:47 +0x34

sync.(*WaitGroup).Wait(0xc420361540)

	/usr/local/go/src/sync/waitgroup.go:131 +0x7a

github.com/weaveworks/cortex/pkg/ruler.TestNotifierSendsUserIDHeader(0xc4205a8d00)

	/go/src/github.com/weaveworks/cortex/pkg/ruler/ruler_test.go:54 +0x2fb

testing.tRunner(0xc4205a8d00, 0x1d64bc8)

	/usr/local/go/src/testing/testing.go:657 +0x96

created by testing.(*T).Run

	/usr/local/go/src/testing/testing.go:697 +0x2ca



goroutine 78 [IO wait]:

net.runtime_pollWait(0x7f4c3b158890, 0x72, 0x0)

	/usr/local/go/src/runtime/netpoll.go:164 +0x59

net.(*pollDesc).wait(0xc4201e5098, 0x72, 0x0, 0xc4201a5de0)

	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x38

net.(*pollDesc).waitRead(0xc4201e5098, 0xffffffffffffffff, 0x0)

	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x34

net.(*netFD).accept(0xc4201e5030, 0x0, 0x293a6a0, 0xc4201a5de0)

	/usr/local/go/src/net/fd_unix.go:430 +0x1e5

net.(*TCPListener).accept(0xc42000e1f0, 0xc420494de0, 0x1a2cd60, 0x298ae40)

	/usr/local/go/src/net/tcpsock_posix.go:136 +0x2e

net.(*TCPListener).Accept(0xc42000e1f0, 0xc420494db0, 0x1a2cd60, 0x298ae40, 0x1b642e0)

	/usr/local/go/src/net/tcpsock.go:228 +0x49

net/http.(*Server).Serve(0xc420078fd0, 0x29498a0, 0xc42000e1f0, 0x0, 0x0)

	/usr/local/go/src/net/http/server.go:2643 +0x228

net/http/httptest.(*Server).goServe.func1(0xc420019800)

	/usr/local/go/src/net/http/httptest/server.go:235 +0x6d

created by net/http/httptest.(*Server).goServe

	/usr/local/go/src/net/http/httptest/server.go:236 +0x5c



goroutine 79 [select]:

github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/discovery.(*TargetSet).Run(0xc420512af0, 0x7f4c3b158950, 0xc420017500)

	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/discovery/discovery.go:192 +0x2e4

created by github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier.(*Notifier).ApplyConfig

	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier/notifier.go:198 +0x3dc



goroutine 82 [select]:

github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier.(*Notifier).Run(0xc420063100)

	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier/notifier.go:240 +0x1bf

created by github.com/weaveworks/cortex/pkg/ruler.(*Ruler).getOrCreateNotifier

	/go/src/github.com/weaveworks/cortex/pkg/ruler/ruler.go:236 +0x1c6

FAIL	github.com/weaveworks/cortex/pkg/ruler	60.089s
```

**Comment by tomwilkie** [comment]
And again:

```
panic: test timed out after 1m0s



goroutine 33 [running]:

testing.startAlarm.func1()

	/usr/local/go/src/testing/testing.go:1023 +0xf9

created by time.goFunc

	/usr/local/go/src/time/sleep.go:170 +0x44



goroutine 1 [chan receive]:

testing.(*T).Run(0xc420587520, 0x1d1b432, 0x1d, 0x1d77570, 0xc420627d20)

	/usr/local/go/src/testing/testing.go:698 +0x2f4

testing.runTests.func1(0xc420587520)

	/usr/local/go/src/testing/testing.go:882 +0x67

testing.tRunner(0xc420587520, 0xc420627de0)

	/usr/local/go/src/testing/testing.go:657 +0x96

testing.runTests(0xc420303200, 0x29b6080, 0xb, 0xb, 0xc420355960)

	/usr/local/go/src/testing/testing.go:888 +0x2c1

testing.(*M).Run(0xc420627f20, 0xc420627f20)

	/usr/local/go/src/testing/testing.go:822 +0xfc

main.main()

	github.com/weaveworks/cortex/pkg/ruler/_test/_testmain.go:62 +0xf7



goroutine 17 [syscall, 1 minutes, locked to thread]:

runtime.goexit()

	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1



goroutine 7 [chan receive]:

github.com/weaveworks/cortex/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x29c1ba0)

	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:882 +0x7a

created by github.com/weaveworks/cortex/vendor/github.com/golang/glog.init.1

	/go/src/github.com/weaveworks/cortex/vendor/github.com/golang/glog/glog.go:410 +0x21d



goroutine 48 [select]:

github.com/weaveworks/cortex/pkg/util.(*hashBucketHistogram).loop(0xc420277720)

	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:87 +0x1e5

created by github.com/weaveworks/cortex/pkg/util.NewHashBucketHistogram

	/go/src/github.com/weaveworks/cortex/pkg/util/hash_bucket_histogram.go:64 +0x18a



goroutine 74 [semacquire]:

sync.runtime_Semacquire(0xc4203559bc)

	/usr/local/go/src/runtime/sema.go:47 +0x34

sync.(*WaitGroup).Wait(0xc4203559b0)

	/usr/local/go/src/sync/waitgroup.go:131 +0x7a

github.com/weaveworks/cortex/pkg/ruler.TestNotifierSendsUserIDHeader(0xc4205875f0)

	/go/src/github.com/weaveworks/cortex/pkg/ruler/ruler_test.go:54 +0x2fb

testing.tRunner(0xc4205875f0, 0x1d77570)

	/usr/local/go/src/testing/testing.go:657 +0x96

created by testing.(*T).Run

	/usr/local/go/src/testing/testing.go:697 +0x2ca



goroutine 75 [IO wait]:

net.runtime_pollWait(0x7f49fa9dab18, 0x72, 0x0)

	/usr/local/go/src/runtime/netpoll.go:164 +0x59

net.(*pollDesc).wait(0xc42024cca8, 0x72, 0x0, 0xc420322020)

	/usr/local/go/src/net/fd_poll_runtime.go:75 +0x38

net.(*pollDesc).waitRead(0xc42024cca8, 0xffffffffffffffff, 0x0)

	/usr/local/go/src/net/fd_poll_runtime.go:80 +0x34

net.(*netFD).accept(0xc42024cc40, 0x0, 0x29587a0, 0xc420322020)

	/usr/local/go/src/net/fd_unix.go:430 +0x1e5

net.(*TCPListener).accept(0xc42000e1d8, 0xc4203de0f0, 0x1a3bfc0, 0x29a9aa0)

	/usr/local/go/src/net/tcpsock_posix.go:136 +0x2e

net.(*TCPListener).Accept(0xc42000e1d8, 0xc4203de0c0, 0x1a3bfc0, 0x29a9aa0, 0x1b75600)

	/usr/local/go/src/net/tcpsock.go:228 +0x49

net/http.(*Server).Serve(0xc420247130, 0x2967de0, 0xc42000e1d8, 0x0, 0x0)

	/usr/local/go/src/net/http/server.go:2643 +0x228

net/http/httptest.(*Server).goServe.func1(0xc4204320c0)

	/usr/local/go/src/net/http/httptest/server.go:235 +0x6d

created by net/http/httptest.(*Server).goServe

	/usr/local/go/src/net/http/httptest/server.go:236 +0x5c



goroutine 76 [select]:

github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/discovery.(*TargetSet).Run(0xc420328190, 0x7f49fa9dabd8, 0xc420518240)

	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/discovery/discovery.go:192 +0x2e4

created by github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier.(*Notifier).ApplyConfig

	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier/notifier.go:198 +0x3dc



goroutine 77 [select]:

github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier.(*Notifier).Run(0xc42019a380)

	/go/src/github.com/weaveworks/cortex/vendor/github.com/prometheus/prometheus/notifier/notifier.go:240 +0x1bf

created by github.com/weaveworks/cortex/pkg/ruler.(*Ruler).getOrCreateNotifier

	/go/src/github.com/weaveworks/cortex/pkg/ruler/ruler.go:236 +0x1c6

FAIL	github.com/weaveworks/cortex/pkg/ruler	60.215s
```

https://circleci.com/gh/tomwilkie/cortex/32?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by jml** [comment]
Fixed by @bboreham in #617
